block/manager.go

```diff
@@ -41,6 +43,9 @@ type Manager struct {
 	State    *types.State
 	Executor *Executor
 
+	// Context
+	cancelCtx context.CancelFunc
```

Comment:
This is a complete abuse / anti-pattern of how to write go code and how to use contexts

https://stackoverflow.com/questions/70042646/best-practices-on-go-context-cancelation-functions

You need to do something like instead https://stackoverflow.com/a/70042766

Sorry but this is just wrong golang

Reply:
Sure there are more idiomatic ways to do this, but I think calling it "complete abuse" of how to write go code is a bit out of line. 
 
Given the time constraints, I don't think these are the most critical issues to deal with right now.